packages/wrapped-keys/src/lib/wrapped-keys.ts

```diff
@@ -0,0 +1,216 @@
+import { decryptToString, encryptString } from '@lit-protocol/encryption';
+import { logError } from '@lit-protocol/misc';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from './utils';
+import {
+  LitMessage,
+  LitTransaction,
+  ExportPrivateKeyParams,
+  ExportPrivateKeyResponse,
+  ImportPrivateKeyParams,
+  ImportPrivateKeyResponse,
+  SignWithEncryptedKeyParams,
+} from './interfaces';
+
+export async function importPrivateKey({
+  pkpSessionSigs,
+  privateKey,
+  litNodeClient,
+}: ImportPrivateKeyParams): Promise<string> {
+  const firstSessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(firstSessionSig);
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: allowPkpAddressToDecrypt,
+      dataToEncrypt: privateKey,
+    },
+    litNodeClient
+  );
+
+  const data = {
+    ciphertext,
+    dataToEncryptHash,
+  };
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const errorBody = await response.text();
+      logError(
+        `Could not import the encrypted key due to the error: ${errorBody}`
+      );
+
+      throw new Error(errorBody);
+    }
+
+    const importedPrivateKey: ImportPrivateKeyResponse = await response.json();
+    return importedPrivateKey.pkpAddress;
+  } catch (error) {
+    const errorMessage = `There was a problem fetching from the database: ${error}`;
+    console.error(errorMessage);
+
+    throw new Error(errorMessage);
+  }
+}
+
+export async function exportPrivateKey({
+  pkpSessionSigs,
+  litNodeClient,
+}: ExportPrivateKeyParams): Promise<string> {
+  const firstSessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(firstSessionSig);
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+    });
+
+    if (!response.ok) {
+      const errorBody = await response.text();
+      logError(
+        `Could not fetch the encrypted key due to the error: ${errorBody}`
+      );
+
+      throw new Error(errorBody);
+    }
+
+    const exportedPrivateKeyData: ExportPrivateKeyResponse =
+      await response.json();
+
+    const decryptedPrivateKey = await decryptToString(
+      {
+        accessControlConditions: allowPkpAddressToDecrypt,
+        chain: CHAIN_ETHEREUM,
```

Comment:
Should we expose this param to the end user?

Reply:
Only for exporting the key not for signing

Reply:
Can this be any key type? if so wouldn't we want to specify `Solana Chains` for example.

Reply:
The chain info is used here just to check the PKP address which is an Eth address